tests/conditional_compilation/test_infer_tool.py

```diff
@@ -0,0 +1,58 @@
+import argparse
```

Comment:
The file name should not start with `test` - those names are reserved for test cases/suites. Please also have a separate folder for the tool

./tests/conditional_compilation/tools/infer_tool.py